internal/prebuiltconfigs/tools/alloydb-postgres-admin.yaml

```diff
@@ -115,9 +115,57 @@ tools:
         type: string
         description: "The full resource name of the primary cluster for a SECONDARY instance. Required only if instanceType is SECONDARY. Otherwise don't ask"
         default: ""
+  alloydb-create-user:
+    kind: http
+    source: alloydb-api-source
+    method: POST
+    path: /v1/projects/{{.projectId}}/locations/{{.locationId}}/clusters/{{.clusterId}}/users
+    description: "Creates a new database user in an AlloyDB cluster. Takes the new user's name and a secure password. Optionally, a list of database roles can be assigned."
+    pathParams:
+      - name: projectId
+        type: string
+        description: "The GCP project ID."
+      - name: locationId
+        type: string
+        description: "The location of the cluster (e.g., 'us-central1')."
+        default: us-central1
+      - name: clusterId
+        type: string
+        description: "The ID of the cluster where the user will be created."
+    queryParams:
+      - name: userId
+        type: string
+        description: "The name for the new user. Must be unique within the cluster."
+    requestBody: |
+      {
+        "userType": "{{.userType}}"
+        {{- if eq .userType "ALLOYDB_BUILT_IN" -}}
+        , "password": "{{.password}}"
+        {{- end -}}
+        {{- if .databaseRoles }}
+        , "databaseRoles": [{{range $i, $role := .databaseRoles}}{{if $i}},{{end}}"{{$role}}"{{end}}]
+        {{- end }}
+      }
+    bodyParams:
+      - name: password
+        type: string
+        description: "A secure password for the new user. Required only for ALLOYDB_BUILT_IN userType."
+        required: false
+      - name: databaseRoles
```

Comment:
How much have you tested this? I worry this is brittle and the tools needs better validation or description to add this array properly